source/kraken/realtime.cpp

```diff
@@ -281,6 +247,69 @@ static bool is_added_stop_time(const transit_realtime::TripUpdate& trip_update)
     return false;
 }
 
+static bool is_handleable(const transit_realtime::TripUpdate& trip_update,
+                          nt::PT_Data& pt_data,
+                          const bool is_realtime_add_enabled,
+                          const bool is_realtime_add_trip_enabled)
+{
+    namespace bpt = boost::posix_time;
+
+    auto log = log4cplus::Logger::getInstance("realtime");
+
+    // if is_realtime_add_enabled = false, this disables the adding trip option too
+    if ((! is_realtime_add_enabled && is_added_stop_time(trip_update)) ||
+        (! is_realtime_add_enabled && is_added_trip(trip_update))) {
+        LOG4CPLUS_DEBUG(log, "Add stop time is disabled: ignoring trip update id "
+                << trip_update.trip().trip_id()
+                << " with ADDED/ADDED_FOR_DETOUR stop times or ADDITIONAL_SERVICE effect");
+        return false;
+    }
+    if (! is_realtime_add_trip_enabled && is_added_trip(trip_update)) {
+        LOG4CPLUS_DEBUG(log, "Add trip is disabled: ignoring trip update id "
+                << trip_update.trip().trip_id()
+                << " with ADDITIONAL_SERVICE effect");
+        return false;
+    }
+    // Case 1: the trip is cancelled
+    if (trip_update.trip().schedule_relationship() == transit_realtime::TripDescriptor_ScheduleRelationship_CANCELED) {
+        return true;
+    }
+    // Case 2: the trip is not cancelled, but modified (ex: the train's arrival is delayed or the train's
+    // departure is brought forward), we check the size of stop_time_update because we can't find a proper
+    // enum in gtfs-rt proto to express this idea
+    if ((trip_update.trip().schedule_relationship() == transit_realtime::TripDescriptor_ScheduleRelationship_SCHEDULED ||
```

Comment:
As you use it multiple times (3 in this PR, maybe more?), is it possible to extract that, for example in a `bool TripUpdate.is_circulating()` ? It's the most expressive I can think of.

Reply:
It don't mind to have this cond. But if it scratches too much I can do something like that
```
is_circulating(trip_update.trip().schedule_relationship()) {

     if trip_update.trip().schedule_relationship() == trt::TripDescriptor_ScheduleRelationship_SCHEDULED || trip_update.trip().schedule_relationship() == trt::TripDescriptor_ScheduleRelationship_ADDED)&& trip_update.stop_time_update_size())
        return true
    else 
        return false
}
```

Reply:
Yep, you can do it as an external function as well (would take trip_update as argument, as you use also stoptimeupdate).
My main point is to mutualize maintenance of this test and promote future reuse.
Having a "one-liner" or not is just codestyle (I prefer "one-liner" like in the current `if` statement).

Reply:
it's okay for me